packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyActionWithVisuals.tsx

```diff
@@ -57,6 +60,19 @@ export const CalculatedPropertyActionWithVisuals = ({
   const [validator, showValidationMessage] = useValidator();
   const [resolvedHiliteIds, setResolvedHiliteIds] = useState<string[]>([]);
   const [colorProperty, setColorProperty] = useState<boolean>(false);
+  const origPropertyName = calculatedProperty?.propertyName ?? "";
+  const [showSaveValidationModal, setShowSaveValidationModal] = useState<boolean>(false);
+  const [invalidCustomCalcs, setInvalidCustomCalcs] = useState<invalids[]>([]);
+
+  async function checkOutliers(changedPropName: string): Promise<invalids[]> {
+    const accessToken = await getAccessToken();
+    const [customCalcProps] = await Promise.all([
+      mappingClient.getCustomCalculations(accessToken, iModelId, mappingId, group.id),
+    ]);
```

Comment:
This `Promise.all` looks redundant.

Reply:
Made the change to restructure the redundant "promise.all" where ever possible.